packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/TerariumAssetCloneService.java

```diff
@@ -241,36 +241,40 @@ public Project importProject(final String userId, final String userName, final P
 		final Project project = projectService.createProject(projectExport.getProject());
 
 		for (final AssetExport assetExport : projectExport.getAssets()) {
-			final AssetType assetType = assetExport.getType();
+			try {
+				final AssetType assetType = assetExport.getType();
 
-			final ITerariumAssetService terariumAssetService = terariumAssetServices.getServiceByType(assetType);
+				final ITerariumAssetService terariumAssetService = terariumAssetServices.getServiceByType(assetType);
 
-			TerariumAsset asset = assetExport.getAsset();
+				TerariumAsset asset = assetExport.getAsset();
 
-			// upload the files (do this first as the asset creation my use the files)
-			for (final Map.Entry<String, FileExport> entry : assetExport.getFiles().entrySet()) {
-				final String filename = entry.getKey();
-				final FileExport fileExport = entry.getValue();
+				// upload the files (do this first as the asset creation my use the files)
+				for (final Map.Entry<String, FileExport> entry : assetExport.getFiles().entrySet()) {
```

Comment:
Maybe outside the scope of this issue, but, looking at the code here it looks like we'll upload the files of an asset before creating the asset. If we're potentially expecting the creation to fail, do we want to do the file upload after or delete the files on failure so that we don't have orphaned files on s3?

Reply:
For some assets the file needs to exist before we create the asset.